```
In file included from /home/travis/build/alacarte-maps/alacarte/src/server/eval/function_operation_node.cpp:27:
In file included from /home/travis/build/alacarte-maps/alacarte/include/server/parser/color_grammar.hpp:25:
/usr/include/boost/spirit/home/qi/string/tst_map.hpp:74:38: error: invalid use of incomplete type 'boost::spirit::qi::tst_pass_through'
            return find(first, last, tst_pass_through());
                                     ^~~~~~~~~~~~~~~~~~
/usr/include/boost/spirit/home/qi/string/tst_map.hpp:20:12: note: forward declaration of 'boost::spirit::qi::tst_pass_through'
    struct tst_pass_through; // declared in tst.hpp
           ^
[  6%] Building CXX object CMakeFiles/alacarte-obj.dir/src/utils/applications.cpp.o
1 error generated.
make[2]: *** [CMakeFiles/server-obj.dir/src/server/eval/function_operation_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/server-obj.dir/all] Error 2
```

strange stuff… how can this work in gcc but not in clang?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,11 @@ find_package(Boost ${BOOST_MIN_VERSION_REQUIRED} COMPONENTS
 	REQUIRED
 )
 
+# Workaround boosts (version < 1.58) crappy c++11 detection for clang
+if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND BOOST_VERSION VERSION_LESS 1.58)
+	add_definitions(-DBOOST_ASIO_HAS_CLANG_LIBCXX)
+endif()
+
 #find other packages
 IF(WIN32)
 	SET(ENV{FREETYPE_DIR} ${CMAKE_FIND_ROOT_PATH})
@@ -63,7 +68,7 @@ find_package(Doxygen)
 #Enable C++11
 if(CMAKE_VERSION VERSION_LESS 3.1 AND (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" OR
                                        CMAKE_CXX_COMPILER_ID STREQUAL "Clang"))
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --std=c++11")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 else()
     set(CMAKE_CXX_STANDARD 11)
     set(CMAKE_CXX_STANDARD_REQUIRED ON)

--- a/include/server/parser/color_grammar.hpp
+++ b/include/server/parser/color_grammar.hpp
@@ -22,6 +22,7 @@
 #ifndef COLOR_GRAMMAR_HPP
 #define COLOR_GRAMMAR_HPP
 
+#include <boost/spirit/home/qi/string/tst.hpp>
 #include <boost/spirit/home/qi/string/tst_map.hpp>
 
 #include "settings.hpp"

